code/drasil-gool/lib/GOOL/Drasil/LanguageRenderer/PythonRenderer.hs

```diff
@@ -535,7 +537,14 @@ instance DeclStatement PythonCode where
   listDecDef = CP.listDecDef
   arrayDec = listDec
   arrayDecDef = listDecDef
-  constDecDef = varDecDef
+  constDecDef v e = do
+    v' <- zoom lensMStoVS v
+    let n = map toUpper (variableName v')
+        newConst = constant n (pure (variableType v'))
+    available <- varNameAvailable n
+    if available
+      then varDecDef newConst e
+      else error "Cannot safely capitalize constant."
```

Comment:
Do we want this to be an error? We could just warn the user and potentially just use the lowercase version of it (assuming that _it_ is available)…

Reply:
That's a good point.  It would be better to do it that way.  The only thing I'm not sure about is, how do we keep track of whether or not the renderer used the modified name?  When the `constant` function shows up later, it needs to know whether or not to modify the variable name.
- We _could_ have it infer whether the modified or unmodified name was used by checking `varNameAvailable (variableName v')` - if the original name was used, we can probably assume that it was used for the constant in question, else we must have modified it.  This would work, but it feels a little sketchy to me.
- A better way might be to have a mapping from GOOL variable name to rendered variable name floating around in the State.  Then in the `constant` function, we could look up the name to use, and not have to worry about any misplaced assumptions.  This would be a bit of extra work to implement, though.

What are your thoughts?  I'd rather not do option 2 until @JacquesCarette can take a look.  Would it be better to do option 1 for now, as it should be a small change?

Reply:
As a side note, what's the best way to print a warning in Haskell?  I'm not actually sure if there's anywhere else in GOOL that currently prints warnings.  It seems like a shame to wrap all of GOOL in `IO String` just for this, is there a better way?

Reply:
Honestly, I'm thinking that "error vs. warning" may be a variability we'd want to introduce at some point. Since this is likely going to be a bigger change, I would create a new issue for tracking our design decisions and working on it in a separate PR (if at all!)

No idea about your warning question 😅